File: test/language/expressions/cpp/NegationOperatorTest.cpp
#include "gtest/gtest.h"

#include "expressions/negation_operator/NegationOperatorExpression.h"

namespace expressions
{
namespace negation_operator
{

TEST(NumBitsOperatorTest, NegatedValue)
{
    NegationOperatorExpression negationOperatorExpression;
    negationOperatorExpression.setValue(true);
    ASSERT_FALSE(negationOperatorExpression.funcNegatedValue());

    negationOperatorExpression.setValue(false);
    ASSERT_TRUE(negationOperatorExpression.funcNegatedValue());
}

} // namespace negation_operator
} // namespace expressions

File: test/language/expressions/python/NegationOperatorTest.py
import Expressions

class NegationOperatorTest(Expressions.TestCase):
    def testNegatedValue(self):
        negationOperatorExpression = self.api.NegationOperatorExpression(True)
        self.assertEqual(False, negationOperatorExpression.negated_value())

        negationOperatorExpression.value = False
        self.assertEqual(True, negationOperatorExpression.negated_value())
